src/platform/intel/ace/platform.c

```diff
@@ -116,6 +128,12 @@ static void notify_pm_state_entry(enum pm_state state)
 						 SOF_MEM_CAPS_L3,
 						 storage_buffer_size);
 #endif /* CONFIG_ADSP_IMR_CONTEXT_SAVE */
+
+		/* change powe state and check if IPC subsystem is prepared to enter D3 */
+		int ret = pm_device_action_run(INTEL_ADSP_IPC_HOST_DEV, PM_DEVICE_ACTION_SUSPEND);
+
+		if (!ret)
+			power_state_failure_report(ret, true, PM_STATE_SOFT_OFF);
```

Comment:
maybe this is correct, but at least it *looks* suspicious - usually `ret == 0` means success, and here a failure is reported?

Reply:
good catch